Extensions/one_click_postage.js

```diff
@@ -1517,25 +1426,62 @@ XKit.extensions.one_click_postage = new Object({
 		XKit.storage.set("one_click_postage", "already_reblogged", JSON.stringify(XKit.extensions.one_click_postage.already_reblogged));
 	},
 
-	show_error: function(code, state) {
-		var m_word = "reblog";
-		if (state === 1) { m_word = "draft"; }
-		if (state === 2) { m_word = "queue"; }
+	show_error: function(error, state) {
+		const verb = ["reblog", "draft", "queue"][state];
+		let message = "";
+
+		if (error.status) {
+			// HTTP error: response object
+			let response = error;
+
+			message = `The server returned HTTP ${response.status}.<br>`;
+			switch (response.status) {
+				case 200:
+					message += "However, Tumblr has indicated that something is wrong.";
+					break;
+				case 401:
+					message += "This usually means your browser is not sending Referer headers.";
```

Comment:
what's the fix for this? error messages should be actionable to the user.

Reply:
it's pretty rare that a user has disabled referer headers without knowing what they are and/or how they did it, so i think adding a line about contacting support under this suffices?